### PR TITLE
clean up; enable WartRemover

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -87,58 +66,32 @@ lazy val scoverageSettings =
...
-      Wart.TraversableOps,
-      Wart.TryPartial,
-      Wart.Var,
-      Wart.While)
-
-    Compile / compile / wartremoverErrors  ++= errorWarts
-  }
-
+lazy val wartRemoverSettings =
   Seq(
```
Comment (AndyHWChung) [inline build.sbt:70]: I made WartRemover more restrictive - let me know if there's something you think we should have ignored or not ignored

File: app/journey/UpdateCanPayUpfrontController.scala [hunk 2]
```diff
@@ -64,7 +64,7 @@ class UpdateCanPayUpfrontController @Inject() (
   }
 
   private def hasChangedAnswer(journey: Journey.HasCanPayUpfront, latestCanPayUpfront: CanPayUpfront): Boolean =
-    journey.canPayUpfront.value != latestCanPayUpfront.value
+    journey.canPayUpfront.value =!= latestCanPayUpfront.value
```
Comment (barnesjake) [inline app/journey/UpdateCanPayUpfrontController.scala:67]: what is this cats wizardry?

**Review by barnesjake — APPROVED** [review]

File: app/journey/UpdateCanPayUpfrontController.scala (hunk 2 above)
Comment (AndyHWChung) [reply app/journey/UpdateCanPayUpfrontController.scala:67]: one of the warts that I enabled was `Equals`. Basically `==` and `!=` isn't typesafe cos you can have different types at either end of an `==` or `!=` and the compiler will not complain. There isn't an equivalent operator inbuilt into scala which is typesafe so I used cats `Eq` typeclass here to help (see [here](https://typelevel.org/cats/typeclasses/eq.html).  

`import cats.syntax.eq._` brings the `===` or `=!=` operators into scope. And you then just need to bring in an implicit `Eq` instance of whatever doing the equals on into scope. There are inbuilt ones in cats for basic types - for our own types we can create our own easily as I've done [here](https://github.com/hmrc/essttp-backend/pull/16/files#diff-20cd501cffb94688a212321b98e1fc0e7a9ece7cad00e82e4dae1171e3dc50cfR30)

Comment (barnesjake) [reply app/journey/UpdateCanPayUpfrontController.scala:67]: Ah, very good to know, thanks!

**Pull request merged**